pyccel/codegen/printing/ccode.py

```diff
@@ -108,11 +108,13 @@ def _traverse_matrix_indices(self, mat):
 
     def _print_PythonFloat(self, expr):
         value = self._print(expr.arg)
-        return '(double)({0})'.format(value)
+        type_name = self.find_in_dtype_registry('real', default_precision['real'])
+        return '({0})({1})'.format(type_name, value)
 
     def _print_Int(self, expr):
         value = self._print(expr.arg)
-        return '(long)({0})'.format(value)
+        type_name = self.find_in_dtype_registry('int', default_precision['int'])
+        return '({0})({1})'.format(type_name, value)
```

Comment:
I wonder if class `ast.builtins.Int` should really be called `PythonInt` to mimic the distinction that we currently have between `ast.numbers.Float` and `ast.builtins.PythonFloat`.

Reply:
It looks like most of the time we are writing `from ast.builtins import Int as PythonInt` anyway so I think that's a good idea

Reply:
Thanks!